R/linear_pool_sample.R

```diff
@@ -42,15 +35,77 @@ linear_pool_sample <- function(model_out_tbl, weights = NULL,
     )
     weights_col_name <- "weight"
   }
+  weight_by_cols <- colnames(weights)[colnames(weights) != weights_col_name]
   unique_weights <- unique(weights[[weights_col_name]])
 
-  if (length(unique_weights) != 1 || length(unique_provided_samples) != 1 || !is.null(n_output_samples)) {
-    cli::cli_abort(
-      "The requested ensemble calculation doesn't satisfy all conditions:
-      1) {.arg model_out_tbl} contains the same number of samples from each component model,
-      2) {.arg weights} are {.val NULL} or equal for every model,
-      3) {.arg n_output_samples} = {.val NULL}"
-    )
+  if (length(unique_weights) != 1) {
+    cli::cli_abort("{.arg weights} must be {.val NULL} or equal for every model")
+  }
+
+  if (!is.null(n_output_samples)) {
+    samples_per_combo <- model_out_tbl |>
```

Comment:
I would be inclined to pull out all of the logic from lines 46 - 88 into a separate function called something like `get_samples_per_combo`.  This is a fairly lengthy and conceptually distinct chunk of code.